forte/data/data_store.py

```diff
@@ -157,12 +164,22 @@ def __init__(
 
         """
         The ``_type_attributes`` is a private dictionary that provides
-        ``type_name``, their parent entry, and the order of corresponding attributes.
-        The keys are fully qualified names of every type; The value is a
-        dictionary with two keys. Key ``attribute`` provides an inner dictionary
-        with all valid attributes for this type and the indices of attributes
-        among these lists. Key ``parent_class`` is a string representing the
-        ancestors of this type.
+        ``type_name``, their parent entry, and the metadata of its corresponding
+        attributes. This metadata includes the order and type information of attributes
```

Comment:
Feels like it would be better to introduce key and value before explaining the metadata of attributes.

Reply:
So key and value here refer to the components of Python `dict`. Do you mean mentioning that they key-value refer to that of a Python `dict`?

Reply:
I'm just thinking moving this sentence `The keys are fully qualified ...` before `This metadata includes the order ...`

Reply:
Got it

Reply:
The docstring still looks a little bit off.

> The ``_type_attributes`` is a private dictionary that provides ``type_name``, ie. their parent entry (represented by the fully qualified name of the entry) as the key, and the

The keys are not parent_entry. And ``type_name`` is different from ``their parent entry``.

Reply:
How about - "The ``_type_attributes`` is a private dictionary that contains the parent entry ( (represented by the fully qualified name of the entry) as key and the metadata ...." ?

Reply:
I still don't think `parent entry` is a key of `_type_attributes`. `parent_class` may be a key of `_type_attributes[type_name]` but it's not a key for `_type_attributes`.